io_scene_niftools/modules/nif_import/animation/transform.py

```diff
@@ -122,37 +154,32 @@ def import_controller_sequence(self, kf_root, b_armature_obj, bind_data):
             # fallout (node_name) & Loki (StringPalette)
             if not n_name:
                 n_name = controlledblock.get_node_name()
-            bone_name = block_registry.get_bone_name_for_blender(n_name)
-            if bone_name not in b_armature_obj.data.bones:
-                continue
-            b_bone = b_armature_obj.data.bones[bone_name]
+            b_target = self.get_target(b_armature_obj, n_name)
+            # todo - temporarily disabled! should become a custom property on both object and pose bone, ideally
             # import bone priority
-            b_bone.niftools.priority = controlledblock.priority
-            # import animation
-            if bone_name in bind_data:
-                niBone_bind_scale, niBone_bind_rot_inv, niBone_bind_trans = bind_data[bone_name]
+            # b_target.niftools.priority = controlledblock.priority
+            # fallout, Loki
+            kfc = controlledblock.interpolator
+            if not kfc:
                 # ZT2
                 kfc = controlledblock.controller
-                # fallout, Loki
-                if not kfc:
-                    kfc = controlledblock.interpolator
-                if kfc:
-                    self.import_keyframe_controller(kfc, b_armature_obj, bone_name, niBone_bind_scale,
-                                                    niBone_bind_rot_inv, niBone_bind_trans)
-        # fallout: set global extrapolation mode here (older versions have extrapolation per controller)
-        if kf_root.cycle_type:
-            extend = self.get_extend_from_cycle_type(kf_root.cycle_type)
-            self.set_extrapolation(extend, b_action.fcurves)
-
-    # TODO [animation] Is scale param required or can be removed, not used
-    def import_keyframe_controller(self, n_kfc, b_obj, bone_name=None, n_bone_bind_scale=None, n_bone_bind_rot_inv=None, n_bone_bind_trans=None):
+            if kfc:
+                actions.add(self.import_keyframe_controller(kfc, b_armature_obj, b_target, b_action_name))
+        for b_action in actions:
+            if b_action:
+                self.import_text_keys(kf_root, b_action)
+                # fallout: set global extrapolation mode here (older versions have extrapolation per controller)
+                if kf_root.cycle_type:
+                    extend = self.get_extend_from_cycle_type(kf_root.cycle_type)
+                    self.set_extrapolation(extend, b_action.fcurves)
 
-        NifLog.debug('Importing keyframe controller for'+b_obj.name)
-
-        b_action = b_obj.animation_data.action
-
-        if bone_name:
-            b_obj = b_obj.pose.bones[bone_name]
+    def import_keyframe_controller(self, n_kfc, b_armature, b_target, b_action_name):
+        """
+        b_target: either Object or PoseBone
```

Comment:
Would keep to docstring to one line

Reply:
Wanted to add more lines, got distracted lol

Reply:
😆